### PR TITLE
Support a stable repository structure for publishing builds

### PR DESCRIPTION
**Pull request by merks**
Reuse JustJ's repository publishing infrastructure.

Ensure that only main branch builds are signed and promoted.

https://github.com/eclipse-mylyn/org.eclipse.mylyn/issues/116

### CODE REVIEW

**Comment by merks** [comment]
@ruspl-afed 

I did some minimal testing that the Jenksfile works in principle.  The build will not actually test the Jenkinsfile changes because it's not trusted in a PR:
```
‘Jenkinsfile’ has been modified in an untrusted revision
```

The promotion will also not be tested because it's in a profile that's not activated.

So this cannot be tested until this is on the main branch...

Note that main branch builds will by default promote while PR builds will **never** promote and also will no longer sign, which will make them faster.  The default build type is nightly...

The builds should be published to the new `updates` folder of this folder:

https://download.eclipse.org/justj/?file=mylyn

Let's review this content once there is content there.

I'll be going out for a while but will be back later in the afternoon.

**Comment by merks** [comment]
@ruspl-afed

Note that if you have not incremented any feature versions yet, we could, if you guys want, go through the entire process of producing a release repository in this new structure, after we've reviewed that a published nightly build actually looks correct...

File: org.eclipse.mylyn-site/pom.xml [hunk 1]
```diff
@@ -25,4 +25,91 @@
...
+                    -data @None
+                    -nosplash
+                    ${org.eclipse.justj.p2.manager.args}
+                    -retain 5
+                    -label "Mylyn"
+                    -build-url ${org.eclipse.justj.p2.manager.build.url}
+                    -root ${project.build.directory}/mylyn-sync
+                    -relative ${org.eclipse.justj.p2.manager.relative}
+                    -version-iu org.eclipse.mylyn.sdk_feature.
+                    -iu-filter-pattern org.eclipse.mylyn.*
```
Comment (ruspl-afed) [inline org.eclipse.mylyn-site/pom.xml:95]: will it allow required 3rd parties to be published as well?

File: org.eclipse.mylyn-site/pom.xml [hunk 2]
```diff
@@ -25,4 +25,91 @@
...
   <name>Mylyn for Eclipse 4.x</name>
+
+  <properties>
+    <eclipse.repo>https://download.eclipse.org/releases/2023-03</eclipse.repo>
+    <justj.tools.repo>https://download.eclipse.org/justj/tools/updates/nightly/latest</justj.tools.repo>
+    <org.eclipse.storage.user>genie.mylyn</org.eclipse.storage.user>
+    <org.eclipse.justj.p2.manager.args>-remote ${org.eclipse.storage.user}@projects-storage.eclipse.org:/home/data/httpd/download.eclipse.org/mylyn</org.eclipse.justj.p2.manager.args>
+    <org.eclipse.justj.p2.manager.extra.args></org.eclipse.justj.p2.manager.extra.args>
+    <org.eclipse.justj.p2.manager.relative>updates</org.eclipse.justj.p2.manager.relative>
+    <org.eclipse.justj.p2.manager.build.url>http://www.example.com/</org.eclipse.justj.p2.manager.build.url>
```
Comment (ruspl-afed) [inline org.eclipse.mylyn-site/pom.xml:36]: this property value looks strange for me

File: Jenkinsfile [hunk 3]
```diff
@@ -11,11 +11,59 @@ pipeline {
...
+						mvn \
+						clean \
+						verify \
+						-B \
+						$MAVEN_PROFILES \
+						-Dmaven.repo.local=$WORKSPACE/.m2/repository \
+						-Dmaven.test.failure.ignore=true \
+						-Dmaven.test.error.ignore=true \
+						-Ddash.fail=false \
+						-Dorg.eclipse.storage.user=genie.cbi \
```
Comment (ruspl-afed) [inline Jenkinsfile:61]: Not sure why do we need to override the user here

**Review by ruspl-afed — APPROVED** [review]

**Comment by ruspl-afed** [comment]
> So this cannot be tested until this is on the main branch...

ok, let's try it in action

**Pull request merged**